integreat_cms/cms/models/abstract_content_translation.py

```diff
@@ -586,6 +595,49 @@ def get_all_used_slugs(self) -> Iterable[str]:
         """
         return self.all_versions.values_list("slug", flat=True)
 
+    def save_new_version(
```

Comment:
Hmm, what is a benefit of that change over the current implementation? 👀

Reply:
The problem I have with the current version of `save_new_version` is that it is really easy to misuse.
The caller needs to make sure that
* `new_translation` has the same language and the same related object as `self` and
* The `version` field of `new_translation` is equal to the version field `self`

We could of course document those requirements, but IMO it would be even better if these requirements to the caller would not exist at all

Reply:
Actually, is there a reason why we need a parameter to that function? If I understand correctly, it would be an option to just accept self and then modify all fields of self, right?

Reply:
I think it'd be a good idea to document this somewhere, I'm not sure if we'll be looking in this PR again to understand this decision again in a few months or years